src/TMap.cpp

```diff
@@ -2183,6 +2183,22 @@ void TMap::downloadMap(const QString& remoteUrl, const QString& localFileName)
         return;
     }
 
+    // Check to ensure we have a map directory to save the map files to.
+    QDir toProfileDir;
+    QString toProfileDirPathString = mudlet::getMudletPath(mudlet::profileMapsPath, mProfileName);
+    if (!toProfileDir.exists(toProfileDirPathString)) {
+        if (!toProfileDir.mkpath(toProfileDirPathString)) {
```

Comment:
ℹ️ Actually, you can take a coding short-cut here and just call the `(bool) QDir::mkPath(const QString &dirPath) const` method in that `if (...) {....}` ***without*** doing the prior `(bool) QDir::exists(const QString &name) const` test first as according to the [Qt Documentation](https://doc.qt.io/qt-5/qdir.html#mkpath):
> #### `bool QDir::mkpath(const QString &dirPath) const`
> Creates the directory path *dirPath*.
>
> The function will create all parent directories necessary to create the directory.
>
> Returns `true` if successful; otherwise returns `false`.
>
> If the path already exists when this function is called, it will return `true`.